ext/scheduler/airflow/dag/template.go

```diff
@@ -0,0 +1,56 @@
+package dag
+
+import (
+	_ "embed"
+	"text/template"
+
+	"github.com/goto/optimus/internal/errors"
+)
+
+//go:embed dag.2.1.py.tmpl
+var dagTemplate214 []byte
+
+//go:embed dag.2.4.py.tmpl
+var dagTemplate243 []byte
+
+type TemplateFactory interface {
+	New(airflowVersion string) *template.Template
+}
+
+type templateFactory struct {
+	tmpl21 *template.Template
+	tmpl24 *template.Template
```

Comment:
can we make this dynamic from embed instead of hard coding here ? we can check if certain version exists or not and through corresponding error